src/main/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/service/BookingCalculationService.kt

```diff
@@ -175,7 +175,33 @@ class BookingCalculationService(
     }
   }
 
+  private fun extractManyTopUpSupervisionDate(booking: Booking): LocalDate? {
+
+    val earliestSentenceDate: LocalDate? = extractionService.earliestDate(
+      booking.sentences, Sentence::sentencedAt
+    )
+
+    val latestUnadjustedExpiryDate: LocalDate? = extractionService.mostRecent(
+      booking.sentences, SentenceCalculation::unadjustedExpiryDate
+    )
+
+    return if (latestUnadjustedExpiryDate != null && earliestSentenceDate != null) {
+      if (latestUnadjustedExpiryDate.isBefore(earliestSentenceDate.plusYears(TWO.toLong())) &&
+        latestUnadjustedExpiryDate.isAfterOrEqualTo(earliestSentenceDate.plusDays(TWO.toLong()))
+      ) {
+        extractionService.mostRecent(
+          booking.sentences, SentenceCalculation::topUpSupervisionDate
+        )
+      } else {
+        null
+      }
+    } else {
+      null
+    }
+  }
+
   companion object {
     val log: Logger = LoggerFactory.getLogger(this::class.java)
+    const val TWO: Int = 2
```

Comment:
Why not just declare this as a Long seen as we are converting it to Long on every use; e.g. 
`const val TWO: Long = 2L`